### DISCUSSION TITLE
Multiple string matchers with an AND operator

### DISCUSSION THREAD

**Discussion by rushter** [projectdiscovery/httpx#1654] [Q&A] [answered]
### httpx version:
v1.6.0

### Current Behavior:
```
echo 'https://github.com/' | httpx --json -ms "Github" and -ms "NON_EXISTENT_PATTERN"

[INF] Current httpx version v1.6.0 (latest)
{"timestamp":"2024-04-03T02:58:46.388201635-05:00",............, "url":"https://github.com/", "status_code":200,"content_length":2236
23,"failed":false,}
```

I was looking at how to add multiple string matchers at the same time and found the comment from one of the developers.
https://github.com/projectdiscovery/httpx/issues/936#issuecomment-1423008837

The problem is that It does not work as expected. It works as an `OR`, not an `AND`.
The above command should not match anything, because the second pattern does not exist.

I'm not sure if this feature even exists. I only found one comment that mentions `and.` Basically, I want to match URLs that match both patterns at the same time.

Sorry, if I just misinterpreted the dev comment.

**Comment by ehsandeep** [comment, marked as answer]
@rushter `echo 'https://github.com/' | httpx --json -ms "Github" -ms "NON_EXISTENT_PATTERN"` is right example to use, possibly not the best example I've shared in linked comment where I was looking for string `and` passed to `-ms` option but you don't need to mention the operator, if you've used multiple `-ms` option, they are treated with AND condition.

```
echo 'https://github.com/' | httpx --json -ms "Github" -ms "NON_EXISTENT_PATTERN"
```

Other ways to do the same with DSL filters like `contains_all`, `contains_any`

```
echo 'https://github.com/' | httpx --json -mdc 'contains_all(body,"GitHub", "XXXXXXX")'
echo 'https://github.com/' | httpx --json -mdc 'contains_any(body,"GitHub", "XXXXXXX")'
```

**Comment by rushter** [comment]
@ehsandeep It looks like it's broken when using the last version of httpx.

```
echo 'https://github.com/' | httpx --json -ms "Github" -ms "NON_EXISTENT_PATTERN"
```
Returns a match.

**Comment by ba1ma0** [comment]
@ehsandeep 

Is this the correct way to write it in the configuration file? If not, please give me a correct demonstration. Thank you.

# filter response with specified string (-fs admin)
filter-string:
  - "The page you are looking for doesn't exist"
  - "Page not found"
  - "This page could not be found"
![image](https://github.com/user-attachments/assets/988374b6-2ff0-46b6-a753-9155f897755b)
